sentry-slog/src/integration.rs

```diff
@@ -1,129 +1,14 @@
-use sentry_core::protocol::{Breadcrumb, Event};
-use sentry_core::{Hub, Integration};
-use slog::{OwnedKVList, Record};
-
-use crate::{breadcrumb_from_record, event_from_record, exception_from_record};
-
-/// The Action that Sentry should perform for a `slog::Level`.
-pub enum LevelFilter {
-    /// Ignore the `Record`.
-    Ignore,
-    /// Create a `Breadcrumb` from this `Record`.
-    Breadcrumb,
-    /// Create a message `Event` from this `Record`.
-    Event,
-    /// Create an exception `Event` from this `Record`.
-    Exception,
-}
-
-/// Custom Mappers
-#[allow(clippy::large_enum_variant)]
-pub enum RecordMapping {
-    /// Adds the `Breadcrumb` to the sentry scope.
-    Breadcrumb(Breadcrumb),
-    /// Captures the `Event` to sentry.
-    Event(Event<'static>),
-}
-
-/// The default slog filter.
-///
-/// By default, an exception event is captured for `critical` logs,
-/// a regular event for `error` and `warning` logs, and breadcrumbs for
-/// everything else.
-pub fn default_filter(level: slog::Level) -> LevelFilter {
-    match level {
-        slog::Level::Critical => LevelFilter::Exception,
-        slog::Level::Error | slog::Level::Warning => LevelFilter::Event,
-        slog::Level::Info | slog::Level::Debug | slog::Level::Trace => LevelFilter::Breadcrumb,
-    }
-}
+use sentry_core::Integration;
 
 /// The Sentry `slog` Integration.
-///
-/// Can be configured with a custom filter and mapper.
-pub struct SlogIntegration {
-    filter: Box<dyn Fn(slog::Level) -> LevelFilter + Send + Sync>,
-    mapper: Option<Box<dyn Fn(&Record, &OwnedKVList) -> RecordMapping + Send + Sync>>,
-}
-
-impl Default for SlogIntegration {
-    fn default() -> Self {
-        Self {
-            filter: Box::new(default_filter),
-            mapper: None,
-        }
-    }
-}
-
-impl std::fmt::Debug for SlogIntegration {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        #[derive(Debug)]
-        struct SlogMapper;
-        let mapper = self.mapper.as_ref().map(|_| SlogMapper);
-        f.debug_struct("SlogIntegration")
-            .field("mapper", &mapper)
-            .finish()
-    }
-}
+#[derive(Debug, Default)]
+pub struct SlogIntegration {}
```

Comment:
I'm starting to wonder if this integration makes sense, since it has no logic. Is there no need for an `slog::` in app filter, at least?

Since this is so basic, and probably also useful even without the integration, what if we remove the integration, just expose the logger, and add the in_app filters to `sentry_core` by default? Same goes for `sentry-log::LogIntegration`.

Reply:
Hm, I’m wondering how useful these "just add myself to the integrations list" kind of integrations are, and if we should just remove those altogether.

Reply:
So I've been thinking a little more about it. If the integration is expected to perform more work during initialization and setup, then I think they do make sense. 

But if they really only add something that could be added by default, then I'd just go ahead and put it in `sentry_core`. The point about the integration is to hook into the lifecycle, but if we don't need that, then we can as well just remove it. 

A good counter example is the panic integration. It registers something during setup. 

cc @HazAT do you have any opinion on this?

Reply:
I think we can make the decision later on if we want to rather remove all these "noop" integrations.

Reply:
Do you mean after this release? Since that would create another case where users have to update their code.

I do not view this as blocking, by the way. I think you can merge as-is, too.

Reply:
we can do it in a followup PR, still in the same release. I think one of the integrations was only recently added anyway.